src/test/java/com/sfuentes/census/ResultCensusTest.java

```diff
@@ -0,0 +1,68 @@
+package com.sfuentes.census;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class ResultCensusTest {
+
+  @Test
+  void getGender_differentGender_ExceptionThrown() {
+    String gender = "Default";
+
+    Exception exception = assertThrows(IllegalArgumentException.class, () ->
+        new Card(1_000, gender, 20, "SINGLE"));
+
+    assertEquals(exception.getMessage(), "Incorrect gender");
+  }
+
+  @Test
+  void getCivilStatus_differentCivilStatus_ExceptionThrown() {
+    String civilStatus = "Free Union";
+
+    Exception exception = assertThrows(IllegalArgumentException.class, () ->
+        new Card(1_000, "FEMALE", 20, civilStatus));
+
+    assertEquals(exception.getMessage(), "Incorrect civil status");
+  }
+
+  @Test
+  void addCensusNumberSingleWomenBetweenSixteenAndTwentyOne_femaleAndMale_upperCase() {
+    var results = new ResultCensus();
+    results.addCensusNumberSingleWomenBetweenSixteenAndTwentyOne(
+        new Card(1, "FEMALE", 16, "SINGLE"));
+    results.addCensusNumberSingleWomenBetweenSixteenAndTwentyOne(
+        new Card(2, "MALE", 16, "SINGLE"));
+
+    assertEquals(List.of(1), results.getCensusNumberCount());
```

Comment:
This is good, but what if the production code changed any other of the properties? here you are only validating the id but not the other properties, this can lead to a problem.

I think this is a good moment to add another dependency to the project is called assertJ:

https://assertj.github.io/doc/#maven

and it has a good assert for this case containsExactlyInAnyOrder
https://www.javadoc.io/doc/org.assertj/assertj-core/latest/org/assertj/core/api/AbstractIterableAssert.html#containsExactlyInAnyOrder(ELEMENT...)

Reply:
Ready, I've changed it.